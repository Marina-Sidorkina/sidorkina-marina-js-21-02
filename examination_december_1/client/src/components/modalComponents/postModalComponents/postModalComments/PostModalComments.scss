.post-modal-comments {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 15px;
  grid-row-gap: 15px;
  justify-content: space-between;
  height: auto;
  min-height: 190px;
  padding: 0;
  margin: 0;
  list-style: none;

  @media (max-width: 788px) {
    grid-template-columns: 1fr;
    height: auto;
  }

  &__info {
    display: flex;
    flex-direction: column;
    width: 100%;
  }

  &__user {
    display: flex;
    align-items: center;
    justify-content: space-between;
    flex-wrap: wrap;

    @media (max-width: 384px) {
      flex-direction: column;
      flex-wrap: nowrap;
      align-items: flex-start;
    }
  }

  &__img {
    width: 40px;
    height: 40px;
    margin-right: 10px;
    border-radius: 50%;
    object-fit: cover;
  }

  &__date {
    display: inline-block;
    font-size: 12px;
    color: #808080;

    @media (max-width: 788px) {
      font-size: 10px;
    }
  }

  &__text {
    margin-top: 5px;

    @media (max-width: 788px) {
      font-size: 12px;
    }
  }

  &__error {
    color: red;
  }

  &__loading {
    color: #1E90FF;
  }
}
