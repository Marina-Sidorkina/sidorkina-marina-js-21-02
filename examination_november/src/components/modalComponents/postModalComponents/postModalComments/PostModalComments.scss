.post-modal-comments {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 15px;
  grid-row-gap: 15px;
  justify-content: space-between;
  height: auto;
  min-height: 190px;
  padding: 0;
  margin: 0;
  list-style: none;

  @media (max-width: 788px) {
    grid-template-columns: 1fr;
    height: auto;
  }

  &__item {
    box-sizing: border-box;
    display: flex;
    height: auto;
    width: 100%;
    min-height: 90px;
    padding: 10px;
    margin: 0 auto;
    border-radius: 15px;
    background-color: #ffffff;
    border: 1px solid #cccccc;

    &_dark {
      background-color: #161b22;
      border: 1px solid #c9d1d9;
      color: #c9d1d9;
    }
  }

  &__info {
    display: flex;
    flex-direction: column;
  }

  &__user {
    display: flex;
    align-items: center;
    justify-content: space-between;
    flex-wrap: wrap;
  }

  &__img {
    width: 40px;
    height: 40px;
    margin-right: 10px;
    border-radius: 50%;
    object-fit: cover;
  }

  &__name {
    margin-bottom: 0;
    margin-right: 5px;
    font-size: 12px;
    font-weight: bold;
    word-break: break-all;
    cursor: pointer;

    @media (max-width: 788px) {
      margin-right: 0;
      font-size: 10px;
    }
  }

  &__date {
    display: inline-block;
    font-size: 12px;
    color: #808080;

    @media (max-width: 788px) {
      font-size: 10px;
    }
  }

  &__text {
    margin-top: 5px;

    @media (max-width: 788px) {
      font-size: 12px;
    }
  }

  &__error {
    color: red;
  }

  &__loading {
    color: #1E90FF;
  }
}
