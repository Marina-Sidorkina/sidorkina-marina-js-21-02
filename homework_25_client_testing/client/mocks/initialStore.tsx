const initialStore = {
  userModal: {
    isOpened: false,
    values: {
      picture: '',
      name: 'John Doe',
      gender: 'male',
      dateOfBirth: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      registerDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      id: '12345',
      email: 'test',
      phone: '55555'
    }
  },
  login: {
    data: {
      error: false,
      isLoading: false,
      inputValue: '',
      authorizedUserId: '123345',
      authorizedUserName: 'Name',
      authorizedUserPicture: ''
    }
  },
  registration: {
    values: {
      name: '',
      gender: '',
      dateOfBirth: '',
      email: '',
      phone: '',
    },
    isLoading: false,
    error: false
  },
  languageSelector: {
    value: 'ru'
  },
  postModal: {
    isOpened: false,
    currentPostId: '12345'
  },
  userPosts: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  postsList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  usersList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      users: [{
        firstName: 'John',
        lastName: 'Doe',
        title: 'mr',
        id: '12345'
      }]
    }
  },
  userInfo: {
    data: {
      isLoading: false,
      user: {
        firstName: 'John',
        lastName: 'Doe',
        picture: '',
        title: 'mr',
        gender: 'male',
        dateOfBirth: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        registerDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        id: '12345',
        email: '',
        phone: '55555'
      }
    }
  },
  postModalPost: {
    isLoading: false,
    error: false,
    owner: {
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    },
    post: {
      id: '12345',
      image: '',
      text: 'text',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }
  },
  postModalComments: {
    page: 1,
    error: false,
    isLoading: false,
    comments: [{
      id: '1234',
      message: 'message',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }],
    owners: [{
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    }],
    limit: 6,
    totalComments: 0
  }
};

export default initialStore;
