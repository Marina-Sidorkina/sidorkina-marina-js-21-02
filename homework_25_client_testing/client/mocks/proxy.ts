import MockAdapter from 'axios-mock-adapter';
import { proxy } from '../src/api/proxy';

export const mock = new MockAdapter(proxy);
