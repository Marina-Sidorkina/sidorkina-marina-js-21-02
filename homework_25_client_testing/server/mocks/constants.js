module.exports = {
  errorText: 'Request failed with status code 520',
  successStatus: 200,
  errorStatus: 520
};
