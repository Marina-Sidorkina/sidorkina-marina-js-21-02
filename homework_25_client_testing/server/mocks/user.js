module.exports = {
  userListMockData: { data: [], page: 0, limit: 5, total: 100 },
  userMockData: {
    id: '60d0fe4f5311236168a109ce',
    title: 'mr',
    firstName: 'Rudi',
    lastName: 'Droste',
    picture: 'https://randomuser.me/api/portraits/med/men/83.jpg',
    gender: 'male',
    email: 'rudi.droste@example.com',
    dateOfBirth: '2021-06-21T21:02:08.029Z',
    phone: '0421-7342787',
    location: {
      street: '1196, MÃ¼hlenweg',
      city: 'GrÃ¼nhain-Beierfeld',
      state: 'Sachsen-Anhalt',
      country: 'Germany',
      timezone: '+3:30'
    },
    registerDate: '2021-06-21T21:02:08.029Z',
    updatedDate: '2021-06-21T21:02:08.029Z'
  },
  userProcessedMockData: {
    id: '60d0fe4f5311236168a109ce',
    firstName: 'Rudi',
    lastName: 'Droste',
    title: 'mr',
    gender: 'male',
    email: 'rudi.droste@example.com',
    phone: '0421-7342787',
    dateOfBirth: {
      enDate: '22 of June 2021',
      ruDate: '22 Июня 2021 года',
      enDateAndTime: '22 of June 00:02',
      ruDateAndTime: '22 Июня 00:02'
    },
    registerDate: {
      enDate: '22 of June 2021',
      ruDate: '22 Июня 2021 года',
      enDateAndTime: '22 of June 00:02',
      ruDateAndTime: '22 Июня 00:02'
    },
    picture: 'https://randomuser.me/api/portraits/med/men/83.jpg'
  },
  userPostsListMockData: {
    data: [{id: '12345', image: 'image', owner: {}, text: 'text'}],
    page: 0, limit: 1, total: 100
  },
  userUpdatedMockData: { firstName: 'Rudi', lastname: 'Droste', gender: 'male',
    email: 'rudi.droste@example.com', dateOfBirth: '2021-06-21T21:02:08.029Z',
    phone: '0421-7342787'
  }
};
