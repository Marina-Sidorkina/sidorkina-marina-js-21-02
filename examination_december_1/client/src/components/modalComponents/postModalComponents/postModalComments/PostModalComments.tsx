import React, { useEffect } from 'react';
import './PostModalComments.scss';
import { useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { getPostModalCommentsListAction } from '../../../../redux/actions/postModalComments';
import { useTypedSelector } from '../../../../redux/hooks/useTypedSelector';
import '../../../../locale/i18next';
import PostModalCommentsItem from '../postModalCommentsItem/PostModalCommentsItem';

const PostModalComments = () => {
  const { t } = useTranslation();
  const stateValues = useTypedSelector((state) => state);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getPostModalCommentsListAction(
      stateValues.postModalComments.page - 1,
      stateValues.postModalComments.limit,
      stateValues.postModal.currentPostId
    ));
  }, [stateValues.postModal.currentPostId, stateValues.postModalComments.page]);

  const getCommentsList = () => {
    if (stateValues.postModalComments.comments.length) {
      return stateValues.postModalComments.comments
        .map((comment, index) => (
          <PostModalCommentsItem
            key={comment.id}
            firstName={stateValues.postModalComments.owners[index].firstName}
            lastName={stateValues.postModalComments.owners[index].lastName}
            id={stateValues.postModalComments.owners[index].id}
            date={comment.publishDate}
            text={comment.message}
            img={stateValues.postModalComments.owners[index].picture}
          />
        ));
    }
    return (
      <div>
        { t('postComments.default', {}) }
      </div>
    );
  };

  const elements = stateValues.postModalComments.isLoading
    ? (
      <div className="post-modal-comments__loading">
        { t('postComments.loading', {}) }
      </div>
    )
    : getCommentsList();

  return (
    <ul className="post-modal-comments">
      { stateValues.postModalComments.error ? (
        <div className="post-modal-comments__error">
          { t('postComments.error', {}) }
        </div>
      ) : elements }
    </ul>
  );
};

export default PostModalComments;
