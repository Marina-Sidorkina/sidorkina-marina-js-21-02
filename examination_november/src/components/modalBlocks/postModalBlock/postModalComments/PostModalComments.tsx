import React, { useContext, useEffect } from 'react';
import './PostModalComments.scss';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { IDummyComment, IDummyOwner } from '../../../../api/dummyApi/@types/dummyApi';
import { processPostsListItemDate } from '../../../../utils/components';
import { getPostModalCommentsListAction } from '../../../../redux/actions/postModalComments';
import { ThemeContext } from '../../../../contexts/ThemeContext';

interface IPostModalCommentsItemProps {
  name: string;
  date: string | null;
  text: string;
  img: string;
}

interface IPostModalCommentsProps {
  comments: IDummyComment[];
  owners: IDummyOwner[];
  isLoading: boolean;
  error: boolean;
  getPostModalCommentsList: Function;
  currentPostId: string;
  limit: number;
  page: number;
}

const PostModalCommentsItem = ({
  name, date, text, img
}: IPostModalCommentsItemProps) => {
  const themeContext = useContext(ThemeContext);

  return (
    <li className={`${themeContext.darkTheme
      ? 'post-modal-comments__item post-modal-comments__item_dark'
      : 'post-modal-comments__item'}`}
    >
      <img
        className="post-modal-comments__img"
        src={img}
        alt="Аватар пользователя"
      />
      <div className="post-modal-comments__info">
        <div className="post-modal-comments__user">
          <div className="post-modal-comments__name">{ name }</div>
          <div className="post-modal-comments__date">{ date }</div>
        </div>
        <p className="post-modal-comments__text">{ text }</p>
      </div>
    </li>
  );
};

const PostModalComments = (props: IPostModalCommentsProps) => {
  const {
    comments, owners, error, isLoading, getPostModalCommentsList, currentPostId,
    limit, page
  } = props;

  useEffect(() => {
    getPostModalCommentsList(page - 1, limit, currentPostId);
  }, [currentPostId, page]);

  const getCommentsList = () => {
    if (comments.length) {
      return comments.map((comment, index) => (
        <PostModalCommentsItem
          key={comment.id}
          name={`${owners[index].firstName} ${owners[index].firstName}`}
          date={processPostsListItemDate(comment.publishDate)}
          text={comment.message}
          img={owners[index].picture}
        />
      ));
    }
    return <div>Комментариев пока нет...</div>;
  };

  const elements = isLoading
    ? <div className="post-modal-comments__loading">Идёт загрузка комментариев...</div>
    : getCommentsList();
  return (
    <ul className="post-modal-comments">
      { error ? <div className="post-modal-comments__error">Ошибка загрузки</div> : elements }
    </ul>
  );
};

export default connect(
  (state: any) => ({
    comments: state.postModalComments.comments,
    owners: state.postModalComments.owners,
    isLoading: state.postModalComments.isLoading,
    error: state.postModalComments.error,
    limit: state.postModalComments.limit,
    page: state.postModalComments.page,
    currentPostId: state.postModal.currentPostId
  }),
  (dispatch) => ({
    getPostModalCommentsList: bindActionCreators(getPostModalCommentsListAction, dispatch)
  })
)(PostModalComments);
