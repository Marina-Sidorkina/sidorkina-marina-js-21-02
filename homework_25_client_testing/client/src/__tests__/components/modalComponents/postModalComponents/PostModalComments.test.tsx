import '../../../../../mocks/matchMediaMock';
import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import PostModalComments
  from '../../../../components/modalComponents/postModalComponents/postModalComments/PostModalComments';
import mockStore from '../../../../../mocks/mockStore';
import initialStore from '../../../../../mocks/initialStore';

configure({ adapter: new Adapter() });

describe('PostModalComments test', () => {
  let store: any;
  let wrapper: any;

  beforeEach(() => {
    store = mockStore(initialStore);
    wrapper = mount(
      <Provider store={store}>
        <HashRouter>
          <PostModalComments />
        </HashRouter>
      </Provider>
    );
  });

  test('should render PostModalComments', () => {
    expect(wrapper.length).toEqual(1);
  });

  test('should render PostModalComments', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
