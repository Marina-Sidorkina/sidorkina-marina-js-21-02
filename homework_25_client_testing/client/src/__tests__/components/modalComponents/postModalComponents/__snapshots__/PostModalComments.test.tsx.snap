// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostModalComments test should render PostModalComments 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <PostModalComments>
        <ul
          className="comments"
        >
          <PostModalCommentsItem
            date={
              Object {
                "ruDateAndTime": "05 October 2011 14:48 UTC",
              }
            }
            firstName="John"
            id="12345"
            img=""
            key="1234"
            lastName="Doe"
            text="message"
          >
            <li
              className="comment"
            >
              <img
                alt="Аватар пользователя"
                className="img"
                src=""
              />
              <div
                className="info"
              >
                <div
                  className="user"
                >
                  <ComponentWithHelper
                    firstName="John"
                    lastName="Doe"
                  >
                    <div
                      className="helper"
                      onMouseOut={[Function]}
                      onMouseOver={[Function]}
                    >
                      <PostModalCommentsItemName
                        firstName="John"
                        lastName="Doe"
                      >
                        <div
                          className="name"
                        >
                          John Doe
                        </div>
                      </PostModalCommentsItemName>
                    </div>
                  </ComponentWithHelper>
                  <div
                    className="date"
                  >
                    05 October 2011 14:48 UTC
                  </div>
                </div>
                <p
                  className="text"
                >
                  message
                </p>
              </div>
            </li>
          </PostModalCommentsItem>
        </ul>
      </PostModalComments>
    </Router>
  </HashRouter>
</Provider>
`;
